src/main/java/com/feedzai/commons/sql/abstraction/dml/Case.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * The copyright of this file belongs to Feedzai. The file cannot be
+ * reproduced in whole or in part, stored in a retrieval system,
+ * transmitted in any form, or by any means electronic, mechanical,
+ * photocopying, or otherwise, without the prior permission of the owner.
+ *
+ * (c) 2014 Feedzai, Strictly Confidential
+ */
+package com.feedzai.commons.sql.abstraction.dml;
+
+import com.google.inject.Injector;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.feedzai.commons.sql.abstraction.dml.dialect.SqlBuilder.k;
+
+/**
+ * Case SQL Expression.
+ *
+ * @author Francisco Santos (francisco.santos@feedzai.com)
+ * @since 2.2.3
+ */
+public class Case extends Expression {
+
+    /**
+     * List of when clauses.
+     */
+    public final List<When> whens;
+
+    /**
+     * Creates an empty case.
+     */
+    protected Case() {
+        this.whens = new ArrayList<>();
+    }
+
+    /**
+     * @return a new case.
+     */
+    public static Case caseWhen() {
+        return new Case();
+    }
+
+    /**
+     * @param condition condition to verify.
+     * @param trueAction action to be executed if the condition is true.
+     * @return a new "case when" that does the trueAction considering the condition.
+     */
+    public static Case caseWhen(final Expression condition, final Expression trueAction) {
+        return caseWhen().when(condition, trueAction);
+    }
+
+    @Override
+    public String translate() {
+        return translator.translate(this);
+    }
+
+    /**
+     * @param condition condition to verify.
+     * @param action action to be executed if the condition is true.
+     * @return this case.
+     */
+    public Case when(final Expression condition, final Expression action) {
+        whens.add(When.when(condition, action));
+        return this;
+    }
+
```

Comment:
Shouldn't you have a `whenElse(...)` method here?  
It seems strange having to instantiate a `CaseElse` if you want to have a else clause.

I would like doing something like:
```
caseWhen(condition, action)
    .when(condition, action)
    .when(condition, action)
    .whenElse(action);
```
Does it make sense?

I know we may not use `case` and `else` because they are keywords.  
But I am thinking that would be prettier to do something like:
```
when(condition, action)
    .when(condition, action)
    .when(condition, action)
    .otherwise(action);
```

Reply:
Wouldn't this force us to write an if statement in the visitor, instead of using polymorphism?
I only created another class to avoid having an if, since the final part of the query changes a little.

Instead of `CASE WHEN ... THEN ... END `we are going to have `CASE WHEN ... THEN ... ELSE ... END`

Is it ok to check if otherwise is null?